cmd/warm-backend-azure.go

```diff
@@ -166,33 +126,30 @@ func newWarmBackendAzure(conf madmin.TierAzure, _ string) (*warmBackendAzure, er
 		return nil, errors.New("no bucket name was provided")
 	}
 
-	if conf.IsSPEnabled() {
-		credential, err = newCredentialFromSP(conf)
-	} else {
-		credential, err = azblob.NewSharedKeyCredential(conf.AccountName, conf.AccountKey)
-	}
-	if err != nil {
-		if _, ok := err.(base64.CorruptInputError); ok {
-			return nil, errors.New("invalid Azure credentials")
-		}
-		return nil, err
+	ep := conf.Endpoint
+	if ep == "" {
+		ep = fmt.Sprintf("https://%s.blob.core.windows.net", conf.AccountName)
 	}
-	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
-	var u *url.URL
-	if conf.Endpoint != "" {
-		u, err = url.Parse(conf.Endpoint)
+
+	if conf.IsSPEnabled() {
+		credential, err := azidentity.NewClientSecretCredential(conf.SPAuth.TenantID, conf.SPAuth.ClientID, conf.SPAuth.ClientSecret, &azidentity.ClientSecretCredentialOptions{})
 		if err != nil {
 			return nil, err
 		}
```

Comment:
Either check `err` after line 139 or use another variable.
```suggestion
		credential, err1 := azidentity.NewClientSecretCredential(conf.SPAuth.TenantID, conf.SPAuth.ClientID, conf.SPAuth.ClientSecret, &azidentity.ClientSecretCredentialOptions{})
		if err1 != nil {
			return nil, err1
		}
```
This is one of the most annoying Go features that variables can be redeclared...